Source/ACE.Server/WorldObjects/Player.cs

```diff
@@ -769,7 +769,7 @@ public bool HandleRunRateUpdate()
         {
             //Console.WriteLine($"{Name}.HandleRunRateUpdates()");
 
-            if (CurrentMovementData.MovementType != MovementType.Invalid)
+            if (CurrentMovementData.MovementType != MovementType.Invalid || CurrentMovementData.Invalid == null)
```

Comment:
this does indeed fix the root issue

as mentioned, this code incorrectly assumed that when MovementType == Invalid, then Invalid would always be populated

Invalid is the default state of MovementType, so this isn't true. this is the correct fix